File: UserSer/UserSer/src/main/java/com/userservice/dto/LoginResponse.java
package com.userservice.dto;

import com.userservice.entity.Role;
import jakarta.persistence.Column;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class LoginResponse {

    private String message;


    private Long id;
    private String name;

    @Column(unique = true, nullable = false)
    private String email;
    private String password;

    @Enumerated(EnumType.STRING)
    private Role role;

    private String profileImage;

    private String state;
    private String district;
    private String taluka;
    private Integer wardNo;


    private LocalDate dob;
    private Integer age;


}

File: citywatch_app/src/app/dtos/loginresponse.ts
export interface LoginResponse {

  message: string;

  id: number;   
  name: string;
  email: string;
  role: string;

  profileImage?: string;

  state: string;
  district: string;
  taluka: string;
  wardNo: number;

  dob: string;   // ISO string (YYYY-MM-DD)
  age: number;
}
